File: sdks/akta-docs/csharp/ConfigModels.cs
using YamlDotNet.Serialization;

namespace OpenAkta.AktaDocs;

public class AktaConfig
{
    [YamlMember(Alias = "schema_version")]
    public string SchemaVersion { get; set; } = "";

    public ProjectCfg Project { get; set; } = new();

    public PathsCfg Paths { get; set; } = new();

    public LinterCfg Linter { get; set; } = new();

    public ScaffoldCfg Scaffold { get; set; } = new();

    public ChangelogCfg Changelog { get; set; } = new();
}

public class ProjectCfg
{
    public string Name { get; set; } = "";

    public string? Slug { get; set; }
}

public class PathsCfg
{
    [YamlMember(Alias = "docs_root")]
    public string DocsRoot { get; set; } = "./akta-docs";

    [YamlMember(Alias = "include_globs")]
    public List<string> IncludeGlobs { get; set; } = new() { "**/*.md" };

    [YamlMember(Alias = "exclude_globs")]
    public List<string> ExcludeGlobs { get; set; } =
        new() { "**/node_modules/**", "**/.git/**", "**/99-archive/**" };
}

public class LinterCfg
{
    [YamlMember(Alias = "default_severity")]
    public string DefaultSeverity { get; set; } = "error";

    public Dictionary<string, RuleOptions> Rules { get; set; } = new();
}

public class RuleOptions
{
    public string? Severity { get; set; }

    [YamlMember(Alias = "min_words")]
    public int? MinWords { get; set; }

    [YamlMember(Alias = "max_words")]
    public int? MaxWords { get; set; }

    [YamlMember(Alias = "min_question_ratio")]
    public double? MinQuestionRatio { get; set; }
}

public class ScaffoldCfg
{
    [YamlMember(Alias = "create_readme_in_each_folder")]
    public bool CreateReadmeInEachFolder { get; set; } = true;

    public bool Gitkeep { get; set; }
}

public class ChangelogCfg
{
    [YamlMember(Alias = "default_target")]
    public string? DefaultTarget { get; set; }

    [YamlMember(Alias = "entry_template")]
    public string EntryTemplate { get; set; } = "compact";

    [YamlMember(Alias = "summary_max_length")]
    public int SummaryMaxLength { get; set; } = 200;
}

File: sdks/akta-docs/java/src/main/java/dev/openakta/aktadocs/ConfigModels.java
package dev.openakta.aktadocs;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
class AktaConfig {
    @JsonProperty("schema_version")
    public String schemaVersion;
    public ProjectCfg project;
    public PathsCfg paths;
    public LinterCfg linter = new LinterCfg();
    public ScaffoldCfg scaffold = new ScaffoldCfg();
    public ChangelogCfg changelog = new ChangelogCfg();
}

@JsonIgnoreProperties(ignoreUnknown = true)
class ProjectCfg {
    public String name;
    public String slug;
}

@JsonIgnoreProperties(ignoreUnknown = true)
class PathsCfg {
    @JsonProperty("docs_root")
    public String docsRoot;
    @JsonProperty("include_globs")
    public List<String> includeGlobs = List.of("**/*.md");
    @JsonProperty("exclude_globs")
    public List<String> excludeGlobs =
            List.of("**/node_modules/**", "**/.git/**", "**/99-archive/**");
}

@JsonIgnoreProperties(ignoreUnknown = true)
class LinterCfg {
    @JsonProperty("default_severity")
    public String defaultSeverity = "error";
    public Map<String, RuleOptions> rules = new HashMap<>();
}

@JsonIgnoreProperties(ignoreUnknown = true)
class RuleOptions {
    public String severity;
    @JsonProperty("min_words")
    public Integer minWords;
    @JsonProperty("max_words")
    public Integer maxWords;
    @JsonProperty("min_question_ratio")
    public Double minQuestionRatio;
}

@JsonIgnoreProperties(ignoreUnknown = true)
class ScaffoldCfg {
    @JsonProperty("create_readme_in_each_folder")
    public boolean createReadmeInEachFolder = true;
    public boolean gitkeep;
}

@JsonIgnoreProperties(ignoreUnknown = true)
class ChangelogCfg {
    @JsonProperty("default_target")
    public String defaultTarget;
    @JsonProperty("entry_template")
    public String entryTemplate = "compact";
    @JsonProperty("summary_max_length")
    public int summaryMaxLength = 200;
}

@JsonIgnoreProperties(ignoreUnknown = true)
class ChangelogEntryPayload {
    @JsonProperty("schema_version")
    public String schemaVersion;
    @JsonProperty("doc_id")
    public String docId;
    public String timestamp;
    @JsonProperty("change_type")
    public String changeType;
    public String summary;
    public String details;
    public String scope;
    public List<String> refs;
}
